src/agentlab/llm/chat_api.py

```diff
@@ -290,10 +279,12 @@ def __init__(
         deployment_name=None,
```

Comment:
try to reconciliate the code of the different inits, may use _post_init() if needed